pkg/archive/archive.go

```diff
@@ -245,14 +245,14 @@ func (ta *tarAppender) addTarFile(path, name string) error {
 	}
 	hdr.Name = name
 
-	nlink, inode, err := setHeaderForSpecialDevice(hdr, ta, name, fi.Sys())
+	inode, err := setHeaderForSpecialDevice(hdr, ta, name, fi.Sys())
 	if err != nil {
 		return err
 	}
 
 	// if it's a regular file and has more than 1 link,
 	// it's hardlinked, so set the type flag accordingly
-	if fi.Mode().IsRegular() && nlink > 1 {
+	if fi.Mode().IsRegular() && hasHardlinks(fi) {
```

Comment:
true. IIRC only directories can not have hardlinks. The size of sockets and symlinks would just be like 4k. So, not terribly skewed for size.